inc/hclib-task.h

```diff
@@ -32,32 +32,20 @@ typedef struct _hclib_task_t {
      * and locality flexible asyncAny
      */
     int is_async_any_type;
```

Comment:
Based off Kumar's comments, I think it would be fine to delete `is_async_any_type`. However, since I don't actually touch that field in these changes, I think that's a separate issue that we can address in a later patch. I don't think this is an issue that should be holding up this pull request.